src/ray/gcs/gcs_server/gcs_health_check_manager.cc

```diff
@@ -158,13 +158,20 @@ void GcsHealthCheckManager::HealthCheckContext::StartHealthCheck() {
   // grpc and io_context contains two different eventloops, here we have to use shared
   // pointer to make sure all memory accesses are valid.
   stub_->async()->Check(
-      &context_,
-      &request_,
-      &response_,
-      [this, start = now, manager = manager](::grpc::Status status) {
+      &context_, &request_, &response_, [this, start = now](::grpc::Status status) {
+        auto manager = manager_.lock();
+        if (manager == nullptr) {
+          RAY_CHECK(stopped_);
```

Comment:
Actually is this always true? I don't see the destructor removes all nodes.

Reply:
https://github.com/ray-project/ray/blob/66cec2f36fcf1582a53fcb54564936150e602f09/src/ray/gcs/gcs_server/gcs_health_check_manager.cc#L208

It's updated to true when health check manager requests to remove node.

Reply:
Yea but `RemoveNode` is not guaranteed to be called when the manager is destructed?

Reply:
Gotcha, removed the check.